src/server/lib/AWS.ts

```diff
@@ -0,0 +1,50 @@
+import * as AWS from 'aws-sdk';
+import { Metrics } from '@/server/models/Metrics';
+import { getConfiguration } from '@/server/lib/configuration';
+import { logger } from '@/server/lib/logger';
+import { AWSError } from 'aws-sdk';
+
+// metric dimensions is a k-v pair
+interface MetricDimensions {
+  [name: string]: string;
+}
+
+const { stage: Stage } = getConfiguration();
+
+const AWS_REGION = 'eu-west-1';
+const PROFILE = 'identity';
+
+const CREDENTIAL_PROVIDER = new AWS.CredentialProviderChain([
+  () => new AWS.SharedIniFileCredentials({ profile: PROFILE }),
+  ...AWS.CredentialProviderChain.defaultProviders,
+]);
+
+const standardAwsConfig = {
+  region: AWS_REGION,
+  credentialProvider: CREDENTIAL_PROVIDER,
+};
+
+const CloudWatch = new AWS.CloudWatch(standardAwsConfig);
+
+export const trackMetric = (
+  metricName: Metrics,
+  dimensions: MetricDimensions = { Stage },
+) =>
+  CloudWatch.putMetricData({
+    Namespace: 'Gateway',
+    MetricData: [
+      {
+        MetricName: metricName,
+        Dimensions: Object.entries(dimensions).map(([Name, Value]) => ({
```

Comment:
in the Scala services we have default dimensions ApiMode (which is the name of the service) and Stage which always get added to any metric sent. It'd be cool to do the same here. The dimensions parameter should probably append to these rather than override them.

Reply:
thanks, i've added the ApiMode, as well as the ability to append to (and override if necessary) the dimensions
https://github.com/guardian/gateway/pull/11/commits/9c71369d5e90fd14117372a572f3c1cc71a3dce4